frontend/genebanks.tsx

```diff
@@ -1,135 +1,66 @@
-  ThirdStateCheck: forwardRef((props, ref) => <Remove {...props} ref={ref} />),
-  ViewColumn: forwardRef((props, ref) => <ViewColumn {...props} ref={ref} />)
-};
+import { Link, useRouteMatch, useHistory } from 'react-router-dom'
+import { GenebankView } from '~genebank_view'
 
-const columns = [
-  {field: 'herd', title: 'Besättning',
-    render: (rowData:any) => <Link to={`/herd/${rowData.herd['herd']}`}>{rowData.herd['herd']}</Link>
-  },
-  {field: 'name', title: 'Namn'},
-  {field: 'certificate', title: 'Certifikat'},
-  {field: 'number', title: 'Nummer',
-    render: (rowData:any) => <Link to={`/individual/${rowData.number}`}>{rowData.number}</Link>
-  },
-  {field: 'sex', title: 'Kön'},
-  {field: 'birth_date', title: 'Födelsedatum'},
-  {field: 'death_date', title: 'Dödsdatum'},
-  {field: 'death_note', title: 'Dödsanteckning'},
-  {field: 'mother', title: 'Moder',
-    render: (rowData:any) => <Link to={`/individual/${rowData.mother['number']}`}>{rowData.mother['name']}</Link>
-  },
-  {field: 'father', title: 'Fader',
-    render: (rowData:any) => <Link to={`/individual/${rowData.father['number']}`}>{rowData.father['name']}</Link>
-  },
-  {field: 'color', title: 'Färg',
-    render: (rowData:any) => rowData.color['name']
-  },
-  {field: 'color_note', title: 'Färganteckning'},
-]
-
-// Define styles for tab menu
+// Define styles
 const useStyles = makeStyles({
-  breadcrumbs: {
-    padding: "15px",
-    paddingBottom: 0,
-  },
-  paper: {
-    height: "calc(100% - 39px)", // remove breadcrumb height
-  },
-  breadcrumb: {
-    textDecoration: "none",
-  },
-  spacer: {
-    height: '39px',
-  },
-  table: {
-    height: "calc(100% - 75px)",
-    padding: "0 20px",
-    overflowY: "scroll",
+  buttonBar: {
+    padding: "5px",
   },
 });
 
 /**
- * Provides genebanks management forms for granting and revoking herd
- * permissions, and managing herd animals.
+ * Provides buttons to select which genebank to view, which is then done with
+ * the GenebankView function.
  */
 export function Genebanks() {
+  const styles = useStyles();
+  const {url} = useRouteMatch()
+  const history = useHistory()
   const {genebanks} = useDataContext()
+  const [subpath, setSubpath] = React.useState('' as String)
   const [genebank, setGenebank] = useState(undefined as Genebank | undefined)
-  const [individuals, setIndividuals] = React.useState([] as Array<Individual>)
-  const styles = useStyles();
 
   React.useEffect(() => {
-    if (genebanks.length > 0 && !genebank) {
-      selectGenebank(genebanks[0].id);
-    }
-  }, [genebanks]);
+    setSubpath(location.pathname.replace(url, '').trim().replace(/\//, ''))
+  })
 
-  const selectGenebank = (id: number) => {
-    if (!genebanks) {
-      return;
-    }
-    const targetGenebank = genebanks.find((g: Genebank) => g.id == id);
-    if (!targetGenebank) {
-      return;
+  React.useEffect(() => {
+    if (!subpath && genebanks.length > 0) {
+      history.push(`${url}/${genebanks[0].name}`)
```

Comment:
Ideally this component should mess around with the location. It would be nice if this was handled somewhere else and the component just got the relevant url pierce as a prop. I don't have a suggestion though so we can skip it for now.

Reply:
Why? doesn't it make sense to have the page relocate to make sure that the actions are available in history?

Reply:
I think browstery history is a great feature, I just wasn't sure it should be in this component. But thinking about it now I realize that this is the navigation component for genebanks, so this should be about right.